.github/workflows/test_on_pr.yml

```
@@ -29,67 +29,81 @@ jobs:
         echo "::error:: Could not start CI tests due to missing *safe PR* label."
         exit 1
     
-  test-mac:
+  test:
     needs: permission
-    runs-on: macos-latest
+    strategy:
+      matrix:
+        os: [windows-latest, ubuntu-latest, macos-latest]
+        size: ['64b', '32b']
+        golang: ['1.18', '1.22']
+        exclude:
+          - os: windows-latest
+            size: '32b'
+          - os: macos-latest
+            size: '32b'
+
+    runs-on: ${{matrix.os}}
     env:
       DBGSYNCLOG: trace
       DBGSYNCON: true
+
     steps:
-    - name: Set up Go ^1.13
-      uses: actions/setup-go@v3
+    - name: Matrix is ${{matrix.os}} / ${{matrix.size}} - Go is ${{matrix.golang}}
+      run: date
+
+    - name: Set up Go ${{matrix.golang}}
+      uses: actions/setup-go@v5
+      if: ${{ matrix.size == '64b' }}
       with:
-        go-version: ^1.13
+        go-version: ${{matrix.golang}}
+        check-latest: false
 
     - name: Check out code into the Go module directory
-      uses: actions/checkout@v3
+      uses: actions/checkout@v4
       with:
         ref: ${{ github.event.pull_request.head.sha }}
+        fetch-depth: 0
 
-    - name: Test without coverage
-      run: make test
-
-  test-windows:
-    needs: permission
-    runs-on: windows-latest
-    env:
-      DBGSYNCLOG: trace
-      DBGSYNCON: true
-    steps:
-    - name: Set up Go ^1.13
-      uses: actions/setup-go@v3
+    - name: Setup Alpine Linux ${{matrix.size}}
+      uses: jirutka/setup-alpine@v1
+      if: ${{ matrix.size == '32b' }}
       with:
-        go-version: ^1.13
+        arch: x86
+        packages: >
+          go
+          make 
+          git 
+          gcc 
+          musl-dev
 
-    - name: Check out code into the Go module directory
-      uses: actions/checkout@v3
-      with:
-        ref: ${{ github.event.pull_request.head.sha }}
+    - name: getconf x86
+      if: ${{ matrix.size == '32b' }}
+      run: |
+        getconf LONG_BIT
+        go version
+      shell: alpine.sh {0}
 
-    - name: Test without coverage
-      run: make test
+    - name: getconf x64
+      if: ${{ matrix.size == '64b' }}
+      run: |
+        getconf LONG_BIT
+        go version
 
-  test-ubuntu:
-    needs: permission
-    runs-on: ubuntu-latest
-    env:
-      DBGSYNCLOG: trace
-      DBGSYNCON: true
-    steps:
-    - name: Set up Go ^1.13
-      uses: actions/setup-go@v3
-      with:
-        go-version: ^1.13
+    - name: Test without coverage (Windows, MacOS)
+      if: ${{matrix.os == 'macos-latest' || matrix.os == 'windows-latest'}}
+      run: make test
 
-    - name: Check out code into the Go module directory
-      uses: actions/checkout@v3
-      with:
-        ref: ${{ github.event.pull_request.head.sha }}
+    - name: Test without coverage (Ubuntu x86)
+      if: ${{matrix.os == 'ubuntu-latest' && matrix.size == '32b'}}
```

Proposition:

```suggestion
      if: ${{matrix.size == '32b'}}
```

It's shorter, but misses some of the context. So I'm not sure it's a good proposition...